### PR TITLE
Fixes #344 - uses registry instead of netsh

### LINKED ISSUE #344: Unable to determine network interface on non-English Windows
When using a Windows with a language different than English, `netsh` output will not be in English.

`lib/landrush/cap/host/windows/configure_visibility_on_host.rb` relies on `netsh` output being in English for it to correctly set the DNS server.

### PR DESCRIPTION
**Pull request by davividal**
When using non-English Windows, netsh output will not be in English.
Using Windows registry is locale independent.

### CODE REVIEW

**Comment by davividal** [comment]
This PR aims to fix #344 .

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # This file needs to be able to execute out of the Vagrant context. Do not require any non core or non relative files
 require 'ipaddr'
+require 'English'
```
Comment (hferentschik) [inline lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Why is that needed?

Comment (davividal) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Because of this:

    lib/landrush/cap/host/windows/configure_visibility_on_host.rb:29:13: C: Style/SpecialGlobalVars: Prefer $CHILD_STATUS from the stdlib 'English' module (don't forget to require it) over $?.
            $?.exitstatus.zero?
            ^^

Comment (hferentschik) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: ahh, got you.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -15,6 +15,12 @@ Metrics/AbcSize:
 # ExcludedMethods: refine
 Metrics/BlockLength:
   Max: 89
+  Exclude:
```
Comment (hferentschik) [inline .rubocop_todo.yml]: So instead of keep adding to the rubocop_todo config, why not try to settle on better defaults. Something like this in _rubocop.yml_ would work for me:

```yaml
Metrics/ModuleLength:
  Max: 200

Metrics/BlockLength:
  Max: 50
  ExcludedMethods: ['describe', 'context']
```

I find the default values to low and I don't think it is worth trying to refactor just to meet these defaults. A max module length of 200 and a max block length of 50 seems reasonable to me. 
Also imo it makes sense to exclude describe and context which would mainly match the minitests. 

If we go down this route, we can remove BlockLength and ModuleLength from the rubocop_todo. WDYT?

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 1 above)
Comment (hferentschik) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: ok

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb [hunk 3]
```diff
@@ -19,46 +20,82 @@ def configure_visibility_on_host(env, ip, tlds)
...
           end
 
-          private
+          # Fetches IP/mask info from registry from a given registry entry
+          # This is a separate method to make testing easier
+          def interface_address(interface)
+            dhcp_enabled = interface.read('EnableDHCP')[1]
+
+            if dhcp_enabled
+              if_ip = interface.read('DhcpIPAddress')[1]
```
Comment (hferentschik) [inline lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Interesting. Did you by any chance test this against a DHCP enabled Windows box as well as one with a static IP?

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb [hunk 4]
```diff
@@ -19,46 +20,82 @@ def configure_visibility_on_host(env, ip, tlds)
...
+              if_mask = interface.read('DhcpSubnetMask')[1]
+            else
+              if_ip = interface.read('IPAddress')[1]
+              if_mask = interface.read('SubnetMask')[1]
+            end
+
+            if if_ip.is_a? Array
```
Comment (hferentschik) [inline lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: What are the elements of the array? Do you know in which cases this occurs?

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb [hunk 5]
```diff
@@ -19,46 +20,82 @@ def configure_visibility_on_host(env, ip, tlds)
...
+              if_ip = if_ip[0]
```
Comment (hferentschik) [inline lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: So the first one is always the right one? Do you have some example interface config?

File: test/landrush/cap/host/windows/configure_visibility_on_host_test.rb [hunk 6]
```diff
@@ -107,31 +65,119 @@ module Windows
           end
         end
 
-        describe '#get_network_name' do
-          it 'returns network name for matching IP' do
-            ConfigureVisibilityOnHost.expects(:`).with('netsh interface ip show config').returns(NETSH_EXAMPLE)
-            expect(ConfigureVisibilityOnHost.get_network_name('192.168.1.193')).must_equal('Ethernet')
+        describe '#get_network_guid' do
```
Comment (hferentschik) [inline test/landrush/cap/host/windows/configure_visibility_on_host_test.rb]: 👍 nice tests

**Comment by hferentschik** [comment]
Good stuff :-)

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 3 above)
Comment (davividal) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Yes, I did. :) That's, actually, how I came across this. :)

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 4 above)
Comment (davividal) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: From what I saw, this only occurs when you have DHCP enabled.
The elements are... a single IP.
I didn't find why Windows stores static IPs as a single-value and DHCP as multi-value in registry.

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 5 above)
Comment (davividal) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Yes, here you have: https://gist.github.com/davividal/0518b75f9fb4fca86f29c6f33373f938

File: .rubocop_todo.yml (hunk 2 above)
Comment (davividal) [reply .rubocop_todo.yml]: I think that lowering BlockLength (from 89 to 50) *and* excluding describe and context are a nicer solution.
For the ModuleLength I think that excluding tests makes more sense than increasing the limit.

**Comment by davividal** [comment]
Yes, I tested this on one Windows box.

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 3 above)
Comment (hferentschik) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Cool

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 4 above)
Comment (hferentschik) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: Hmm, a bit random, but fair enough I guess

File: .rubocop_todo.yml [hunk 7]
```diff
@@ -14,7 +14,12 @@ Metrics/AbcSize:
 # Configuration parameters: CountComments, ExcludedMethods.
 # ExcludedMethods: refine
 Metrics/BlockLength:
-  Max: 89
+  Max: 50
```
Comment (hferentschik) [inline .rubocop_todo.yml]: Should we not move this into `.rubocop.yml`? The idea is to eventually get rid of `.rubocop_todo.yml` , either by refactoring code to not require any non-default configuration or by setting some reasonable project specific configuration in  `.rubocop.yml`.

File: .rubocop_todo.yml [hunk 8]
```diff
@@ -14,7 +14,12 @@ Metrics/AbcSize:
...
+  ExcludedMethods: ['describe', 'context']
+
+Metrics/ModuleLength:
+  Exclude:
+    - !ruby/regexp /^test\//
```
Comment (hferentschik) [inline .rubocop_todo.yml]: Sweet. I did not know that one can do that.

File: lib/landrush/cap/host/windows/configure_visibility_on_host.rb (hunk 4 above)
Comment (davividal) [reply lib/landrush/cap/host/windows/configure_visibility_on_host.rb]: After reading around, it looks like this is like this for a reason: if the interface is not using DHCP, you can have multiple IPs on the same interface.
This makes sense to me. But I don't have a use case for this, so I am not able to properly test. I'm pretty sure that this (multiple IPs on the same NIC) will break things, but I think that this is such an edge case that I shouldn't care about it *right now*. :)

File: .rubocop_todo.yml (hunk 7 above)
Comment (davividal) [reply .rubocop_todo.yml]: Moved.

**Comment by davividal** [comment]
Just squashed the commits.
